M2/Macaulay2/m2/startup.m2.in

```diff
@@ -1,642 +0,0 @@
--- -*- coding: utf-8 -*-
---		Copyright 1993-2009 by Daniel R. Grayson
--- layout.m2, includes the former contents of ../m2/startup.m2
--- this file gets incorporated into the executable file bin/M2 as the string 'startupString'
--- we want to ignore the --datarootdir setting, whatever that means, so here we we mention it: @datarootdir@
-
-interpreterDepth = loadDepth = errorDepth = 0
-debuggingMode = true
-stopIfError = false
-notify = false
-debugWarningHashcode = null
-gotarg := arg -> any(commandLine, s -> s === arg)
-randomSeed = -- initializing this here so the startup is reproducible
-if gotarg "--no-randomize" then 0 else (currentTime() << 16) + processID()
-if gotarg "--notify" then notify = true
-if gotarg "--stop" then stopIfError = true
-if gotarg "--int"  then handleInterrupts = false
-
-firstTime := class PackageDictionary === Symbol
-
-if firstTime then (
-    -- we do this bit *before* "debug Core", so that Core (the symbol, not the package),
-    -- which may not be there yet, ends up in the right dictionary
-    match = x -> null =!= regex x;
-    assert' = x -> (
-	if class x =!= Boolean then error "assert: expected true or false";
-	if not x then error "assertion failed");
-    printerr = msg -> endl(printString(stderr, concatenate(loadDepth, "-- ", msg)));
-    value = value'; -- TODO: is there a different way to get --no-core to work?
-    --
-    PackageDictionary = new Dictionary;
-    dictionaryPath = append(dictionaryPath, PackageDictionary);
-    assert'( not isGlobalSymbol "Core" );
-    PackageDictionary#("Package$Core") = getGlobalSymbol(PackageDictionary, "Core");
-    )
--- we can't make this an else-clause, because then "Core" will be in the wrong dictionary
-disassemble ( () -> debug Core )
-if not firstTime then debug Core -- we need access to the private symbols (we remove the Core private dictionary later.)
-
-toString := value' getGlobalSymbol if firstTime then "simpleToString" else "toString"
-
-local exe
-local topBuilddir
```

Comment:
What's the point of declaring these so early? Why not just say `exe = commandLine#0`?